src/ChannelDetector.ts

```diff
@@ -400,30 +405,36 @@ export class ChannelDetector {
             pattern: Types.unknown,
             usedInCurrentDevice: [],
             state: {} as InternalDetectorState,
+            ignoreEnums: !!options.ignoreEnums,
         };
 
         for (const pattern in patterns) {
-            if (!ChannelDetector.patternIsAllowed(patterns[pattern], options.allowedTypes, options.excludedTypes)) {
+            if (
+                options._checkedPatterns.includes(pattern as Types) ||
+                !ChannelDetector.patternIsAllowed(patterns[pattern], options.allowedTypes, options.excludedTypes)
+            ) {
                 continue;
             }
+            options._checkedPatterns.push(pattern as Types);
 
             context.result = null;
 
             context.pattern = pattern as Types;
             context.usedInCurrentDevice = [];
-            patterns[pattern].states.forEach(state => {
+            for (const state of patterns[pattern].states) {
                 let found = false;
 
                 // one of the following
                 context.state = state;
                 if (this._testOneState(context)) {
+                    console.log(state.name, 'used in current device', context.usedInCurrentDevice);
```

Comment:
I would only use console.log for debug and not in production. Browser console will be filled up with such a messages

Reply:
yes thats not intnded ... i remove it. left over